libkatana_python_native/include/katana/python/Conventions.h

```diff
@@ -0,0 +1,101 @@
+#ifndef KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_CONVENTIONS_H_
+#define KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_CONVENTIONS_H_
+
+#include <type_traits>
+
+#include <pybind11/pybind11.h>
+
+namespace katana {
+
+namespace detail {
+
+// In these classes, the empty implementations are the default cases used when
+// the class T does *not* have the method being checked for.
+
+/// DefRepr will def `__repr__` based on `ToString` if it is available.
+template <typename T, typename Enable = void>
+struct DefRepr {
+  void operator()(pybind11::class_<T>& cls [[maybe_unused]]) {}
+};
+
+template <typename T>
+struct DefRepr<T, std::void_t<decltype(std::declval<T>().ToString())>> {
+  void operator()(pybind11::class_<T>& cls) {
+    cls.def("__repr__", &T::ToString);
+  }
+};
+
+/// DefEqualsEquals will def ``__eq__`` (Python ``==``) based on operator==() if
+/// it is available.
+template <typename T, typename Enable = void>
+struct DefEqualsEquals {
+  void operator()(pybind11::class_<T>& cls [[maybe_unused]]) {}
+};
+
+template <typename T>
+struct DefEqualsEquals<
+    T, std::void_t<decltype(std::declval<T>() == std::declval<T>())>> {
+  void operator()(pybind11::class_<T>& cls) {
+    cls.def("__eq__", &T::operator==);
+  }
+};
+
+/// DefEqualsEquals will def ``__eq__`` (Python ``==``) based on Equals() or
+/// operator==() if one is available (`Equals` is preferred).
```

Comment:
```
/// DefEquals will def ``__eq__`` (Python ``==``) based on Equals() if it is available
/// or defer to DefEqualsEquals
```

Reply:
I prefer to document behavior instead of implementation. But I will add a reference to DefEqualsEquals to help people tracing the implementation.